Tests/UnitTests/TextEditor/ViewModel/TextEditorViewModelTests.swift

```diff
@@ -0,0 +1,267 @@
+//
+//  TextEditorViewModelTests.swift
+//  SparkTextEditorUnitTests
+//
+//  Created by robin.lemaire on 13/11/2023.
+//  Copyright © 2023 Adevinta. All rights reserved.
+//
+
+import Foundation
+import XCTest
+@testable import SparkTextInput
+@_spi(SI_SPI) @testable import SparkTextInputTesting
+@_spi(SI_SPI) import SparkCommon
+@_spi(SI_SPI) import SparkCommonTesting
+@_spi(SI_SPI) import SparkThemingTesting
+import SparkTheming
+import SparkTheme
+import Combine
+
+final class TextEditorViewModelTests: XCTestCase {
+
+    // MARK: - Properties
+
+    private var subscriptions = Set<AnyCancellable>()
+
+    // MARK: - Setup
+
+    override func tearDown() {
+        super.tearDown()
+
+        // Clear publishers
+        self.subscriptions.removeAll()
+    }
+
+    // MARK: - Init Tests
+
+    func test_properties_on_init() {
+        // GIVEN / WHEN
+        let stub = Stub()
+
+        stub.subscribePublishers(on: &self.subscriptions)
+
+        // THEN
+
+        // **
+        // Published properties
+
+        // Should Update Vertical Spacing
+        TextEditorViewModelPublisherTest.XCTAssert(
+            shouldUpdateVerticalSpacing: stub.shouldUpdateVerticalSpacingPublisherMock,
+            expectedNumberOfSinks: 1,
+            expectedValue: 0
+        )
+
+        // Is Placeholder
+        TextEditorViewModelPublisherTest.XCTAssert(
+            isPlaceholder: stub.isPlaceholderPublisherMock,
+            expectedNumberOfSinks: 1,
+            expectedValue: false
+        )
+        // **
+
+        // Use Cases
+        TextEditorGetVerticalSpacingUseCaseableMockTest.XCTCallsCount(
+            stub.getVerticalSpacingUseCaseMock,
+            executeWithHeightAndFontNumberOfCalls: 0
+        )
+    }
+
+    // MARK: - Update
+
+    func test_contentChanged_without_content() {
+        self.testContentChanged(isContent: false)
+    }
+
+    func test_contentChanged_with_content() {
+        self.testContentChanged(isContent: true)
+    }
+
+    func testContentChanged(isContent: Bool) {
```

Comment:
isContent -> hasContent

Reply:
Ok ! Change is on this [branch](https://github.com/adevinta/spark-ios-component-text-input/tree/1087/uikit).